
From: Tom Rini <trini@kernel.crashing.org>

The following patch is needed so that kernel-doc can handle functions which
have __attribute__'s on them (such as __attribute__ ((weak))).


---

 25-akpm/scripts/kernel-doc |    1 +
 1 files changed, 1 insertion(+)

diff -puN scripts/kernel-doc~kerneldoc-handle-attributes scripts/kernel-doc
--- 25/scripts/kernel-doc~kerneldoc-handle-attributes	2004-04-03 02:59:57.240651040 -0800
+++ 25-akpm/scripts/kernel-doc	2004-04-03 02:59:57.244650432 -0800
@@ -1376,6 +1376,7 @@ sub dump_function($$) {
     $prototype =~ s/^inline +//;
     $prototype =~ s/^__inline__ +//;
     $prototype =~ s/^#define +//; #ak added
+    $prototype =~ s/__attribute__ \(\([a-z,]*\)\)//;
 
     # Yes, this truly is vile.  We are looking for:
     # 1. Return type (may be nothing if we're looking at a macro)

_
